src/features/device-state/routes/state.ts

```diff
@@ -362,24 +488,56 @@ const getConfig = (device: AnyObject) => {
 const getUserAppForState = (
 	device: AnyObject,
 	config: Dictionary<string>,
-): Partial<App> => {
+): AppV3 => {
 	const userAppFromApi: AnyObject = device.belongs_to__application[0];
 
 	// get the release of the main app that this device should run...
 	const release = getReleaseForDevice(device);
 
 	// grab the main app for this device...
 	return release == null
-		? {
+		? ({
 				name: userAppFromApi.app_name,
+				appId: userAppFromApi.id,
+				uuid: userAppFromApi.uuid,
 				services: {},
 				networks: {},
 				volumes: {},
-		  }
+		  } as AppV3)
 		: buildAppFromRelease(device, userAppFromApi, release, config);
 };
 
-const getDependent = (device: AnyObject): Dependent => {
+const getSupervisorAppForState = async (
+	device: AnyObject,
+	config: Dictionary<string>,
+): Promise<AppV3 | null> => {
+	const deviceManagedByRelease = device?.should_be_managed_by__release?.[0];
+	const deviceManagedByApplication =
+		deviceManagedByRelease?.belongs_to__application?.[0];
+	if (!deviceManagedByApplication) {
+		return null;
+	}
+
+	const fromApi: AnyObject = {};
+	fromApi.id = deviceManagedByApplication.id;
+	fromApi.app_name = deviceManagedByApplication.app_name;
+	fromApi.uuid = deviceManagedByApplication.uuid;
+	fromApi.application_config_variable =
+		deviceManagedByApplication.application_config_variable;
+	fromApi.application_environment_variable =
+		deviceManagedByApplication.application_environment_variable;
+	fromApi.should_be_running__release = deviceManagedByRelease;
+
+	const supervisorApp = await buildAppFromRelease(
```

Comment:
This `await` is unnecessary right? Can you remove it please
```suggestion
	const supervisorApp = buildAppFromRelease(
```